prover/src/matrix/row_matrix.rs

```diff
@@ -50,25 +52,40 @@ impl<E: FieldElement> RowMatrix<E> {
     ///
     /// To improve performance, polynomials are evaluated in batches specified by the `N` type
     /// parameter. Minimum batch size is 0.
-    pub fn evaluate_polys<const N: usize>(polys: &Matrix<E>, blowup_factor: usize) -> Self {
-        assert!(N > 0, "batch size N must be greater than zero");
```

Comment:
Even though it is triggered in `build_segments()`, we may want to keep the check at the higher level? Not really blocking though.

Reply:
Agreed! I've added this check to all constructors.